02-peer-protocol.md

```diff
@@ -787,9 +787,17 @@ is destined, is described in [BOLT #4](04-onion-routing.md).
 #### Requirements
 
 A sending node:
-  - MUST NOT offer `amount_msat` it cannot pay for in the
-remote commitment transaction at the current `feerate_per_kw` (see "Updating
-Fees") while maintaining its channel reserve.
+  - if it is _responsible_ for paying the Bitcoin fee:
+    - MUST NOT offer `amount_msat` it cannot pay for in the remote commitment
+    transaction at the current `feerate_per_kw` (see "Updating Fees") while
+    maintaining its channel reserve.
+    - MUST NOT offer `amount_msat` if the resulting balance doesn't allow it to
+    pay the fee for a future additional HTLC at `2*feerate_per_kw` while
```

Comment:
This factor 2 should be challenged. Matt proposed 5, Rusty implemented 1.5, meeting in the middle was an ugly 3.25 so I chose 2 completely randomly.

Reply:
I think hard-coding the `2*` via MUST (and later also making this a hard requirement on the receiver side) is not necessary and could be downgraded to SHOULD.

Reply:
I'm ok with making that one a SHOULD. However please note that if we go down the road of this "extra reserve", I think it's important that implementations agree on the factor (current `2`). That allows a node to compute exactly how much the remote is able to send or receive without having to try and fail. Does that make sense?

Reply:
suggestion: 

```
pay the fee for a future additional untrimmed HTLC at `2*feerate_per_kw` while
```

Reply:
Do we have other cases where a remote can't reliably predict if a HTLC is going to be rejected, knowing channel parameters and balance ? If yes, SHOULD is fine.

Reply:
We can never really predict that, because a node can implement any policy it wants on HTLC filtering and reject HTLCs arbitrarily...there's always a chance that a valid HTLC you're offering ends up being rejecting by the remote.

There's also always the risk of a concurrent fee update, which makes your valid-looking HTLC invalid but you couldn't know that because you didn't yet receive information about the new fee.

Reply:
I updated this to a SHOULD.